src/component/GetToken.js

```diff
@@ -0,0 +1,21 @@
+import React from "react";
+
+import getToken from "../token";
+
+const GetToken = () => {
+  // Function triggered on button click which
+  // will create a unique token, save it to
+  // local storage & reload browser
+  const onButtonClickHandler = () => {
+    const uniqueToken = getToken();
+    localStorage.setItem("uniqueToken", uniqueToken);
+    window.location.reload(false);
```

Comment:
I'd say that even though this works (yay!), this is not _the React way_ of doing things, here, you're basically, telling the browser to reload, so it renders the app again, and goes through this `if` statement again

```js
if (checkToken === null) {
    return <GetToken />;
  } else {
    return <Router />;
  }
```
Ideally, we should not trigger a browser refresh to get the app to a new state, we can't see the real drawbacks here since it is a really small example, but imagine that we were filling up a form, and then, we trigger the refresh, the user would lose all its information because of that!

What I'd suggest here, is to `lift the state up` https://reactjs.org/docs/lifting-state-up.html, that means that the app's state is handled in the more outermost component, in this case, the `App` component. I'm thinking something like:

```
1. The user clicks on the Get Token button
2.The user get the new token, and it gets stored in the App's state
3. Because the App's component state changed, it will trigger a re-render
4. That re-render will take the user to the shopping list because the token already exists.
```

Let me know if you have any questions about this!